File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        ans = 0
        one_count_exists = False
        for c, cnt in Counter(s).items():
            if cnt & 1:
                one_count_exists = True
                cnt -= 1
                
            ans += cnt 
        return ans + 1 if one_count_exists else ans

File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        Map<Character, Integer> counts = new HashMap<>();
        for(char c : s.toCharArray()){
            if(!counts.containsKey(c)){
                counts.put(c, 0);
            }
            
            counts.put(c, counts.get(c) + 1);
        }
        
        int ans = 0;
        boolean oddExists = false;
        for(int cnt : counts.values()){
            if(cnt % 2 == 1){
                cnt -= 1;
                oddExists = true;
            }
            
            ans += cnt;
        }
        
        return oddExists ? ans + 1 : ans;
    }
}